### DISCUSSION TITLE
you are dismissed! But still equiped

### DISCUSSION THREAD

**Discussion by RomashinTimofey** [trickerer/Trinity-Bots#835] [Ideas]
Greetings! So my question in title ;] How to do - after Dissmissed bot can save hes manage roles, talant, and equip. that must be like tomogochi ^^' Pls help and ask where need to change code for this <3

**Comment by RomashinTimofey** [comment]
In general, I'm looking for an answer here, but in the meantime I tried to solve the question myself. Spoiler alert - to no avail. I tried to watch what is launched at the dismiss stage of the bot. Something like reset Default and I tried to comment it out - I didn't get the desired result. ideally, do it at all and wrap it in the same way in the world.conf file.

**Reply by Russianranger** [reply]
Well you didn't look hard enough - it is in the documentation. You simply "unbind" the NPCBot which will dismiss them for the session, or until you rebind them. This will retain all settings/gear. You can setup macros to make it easier to execute.

For example - .npcb co unb Gart
That unbinds the bot (based on their name, in this case it's Gart).
To rebind it is - .npcb co reb Gart

You can do this for multiple bots in a single command .npcb co unb Gart Torm etc

Bear in mind - they still are only for the character that hired them and cannot be transferred between characters unless you completely fire them. They will also reappear automatically after you log out/log back in

**Reply by RomashinTimofey** [reply]
Greet thanks for answer!
After reviewing the offer, this is really close to what I want, but there is a lot of hassle with rebind unbind, I would like the bots to be more independent..In general, I'm looking for an answer here, but in the meantime I tried to solve the question myself. Spoiler alert - to no avail. I tried to watch what is launched at the dismiss stage of the bot. Something like reset Default and I tried to comment it out - I didn't get the desired result. ideally, do it at all and wrap it in the same way in the world.conf file.

**Comment by trickerer** [comment]
#55

**Reply by RomashinTimofey** [reply]
One more Great Thanks! But this post still is part of puzzle. I would like them to be static, since the bot has received a level, then it still has this level. Once he was given an object, it means that the object remained to him, and he throws away the previous one and does not return it. That is, an imitation of a player and yes, you can immediately hire a level 80 bot, which would be similar to approaching high lvl in a real game and saying "hi bro, help me in WC - boost me" and after the logout, the bots break away from the binding (this is already implemented, I do it after a while in the logout)

**Reply by Russianranger** [reply]
> One more Great Thanks! But this post still is part of puzzle. I would like them to be static, since the bot has received a level, then it still has this level. Once he was given an object, it means that the object remained to him, and he throws away the previous one and does not return it. That is, an imitation of a player and yes, you can immediately hire a level 80 bot, which would be similar to approaching high lvl in a real game and saying "hi bro, help me in WC - boost me" and after the logout, the bots break away from the binding (this is already implemented, I do it after a while in the logout)

You may be better served by going the "Playerbots" route - this does what you're saying.

What you're trying to do here doesn't have the database tables/code to support the request. You would need to create independent bot tables to monitor levels, then you would need to alter how bots gain levels, because currently bots don't "earn experience" when they are employed by the player. They just copy their level, that's it. The wandering bots can level up, but only retain for that session. By storing all that information, you'll be slowing your server down a lot depending on how many bots you're trying to run - and honestly - playerbots already do these things. They just aren't that great at combat and... slow your server down.

**Reply by trickerer** [reply]
The point is: leaving equipment on a bot when dismissed creates an exploit allowing to transfer soulbind items between players, I'm not doing that.

Bot level isn't saved anywhere, it's just matched with player. You can either set it randomly or save it to DB, altering existing `characters_npcbot` table structure.

In general if you want hire-able pre-equipped bots (without any equipment management) or any other equipment management system but the existing one - some forks and repacks implement that but you'll have to find them yourself. I do understand what you want - this isn't the first time I get this question - but that's out of scope of this project and **requires a lot of code changed**. Besides, bots aren't designed to do everything for you, boost you or remove all game challenge otherwise, the default bots maximum in config is just one for a reason.

**Comment by RomashinTimofey** [comment]
thanks a lot for the answers and involvement in the discussion of the topic. I generally play alone and the server is for myself, so the exploit with the transfer of items is not terrible for me =))) But I realized that it takes a lot of time to change the code. Ideally, if a guru like you had told me on the fly which sections of the code would have to be changed and which tables. So far, it seems that I can't do this again ^^'
